### PR TITLE
refactor(connector): use AwsAuthProps instead of HashMap in config

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

To make the code more clear. And we can see the configs in the yaml

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/connector/with_options_sink.yaml [hunk 1]
```diff
@@ -365,9 +386,30 @@ PulsarConfig:
...
+    required: false
+  - name: session_token
+    field_type: String
+    required: false
+  - name: arn
+    field_type: String
+    required: false
+  - name: external_id
+    field_type: String
+    required: false
```
Comment (xxchan) [inline src/connector/with_options_sink.yaml:410]: Oops, this field is a new one, not included in `AWS_DEFAULT_CONFIG`

Comment (xxchan) [reply src/connector/with_options_sink.yaml:410]: BTW, I noticed there are some inconsistencies: 

For kinesis, these are like
```
  - name: aws.credentials.secret_access_key
    field_type: String
    required: false
    alias: kinesis.credentials.secret
```

But for Bigquery and Pulsar, these are simple. Not sure whether/how to improve it.

File: src/connector/src/common.rs [hunk 2]
```diff
@@ -51,6 +50,111 @@ pub struct AwsPrivateLinkItem {
     pub port: u16,
 }
 
+use aws_config::default_provider::region::DefaultRegionChain;
+use aws_config::sts::AssumeRoleProvider;
+use aws_credential_types::provider::SharedCredentialsProvider;
+use aws_types::region::Region;
+use aws_types::SdkConfig;
+
+/// A flatten cofig map for aws auth.
```
Comment (BugenZhao) [inline src/connector/src/common.rs:59]:
```suggestion
/// A flatten config map for aws auth.
```

File: src/connector/src/common.rs [hunk 3]
```diff
@@ -51,6 +50,111 @@ pub struct AwsPrivateLinkItem {
...
+    fn default() -> Self {
+        Self {
+            region: Default::default(),
+            endpoint: Default::default(),
+            access_key: Default::default(),
+            secret_key: Default::default(),
+            session_token: Default::default(),
+            arn: Default::default(),
+            external_id: Default::default(),
+            profile: Some("default".to_string()),
```
Comment (BugenZhao) [inline src/connector/src/common.rs:83]: Should we also make it a `#[serde(default)]`? Also they can be unified with `serde_default`.

Comment (xxchan) [reply src/connector/src/common.rs:83]: Maybe we should remove the `"default"`, as there are usages like `profile: Default::default(),`....

Comment (xxchan) [reply src/connector/src/common.rs]: image

File: src/connector/with_options_sink.yaml (hunk 1 above)
Comment (tabVersion) [reply src/connector/with_options_sink.yaml:410]: because aws provides more complex auth, I think we miss the `external_id` part for s3 auth.

Comment (xxchan) [reply src/connector/with_options_sink.yaml:410]: Didn't get it. Do you mean `external_id` is needed or not needed? Do we need to change anything in this PR?

Comment (tabVersion) [reply src/connector/with_options_sink.yaml:410]: `external_id` is an optional field for aws auth. IIRC, we impl it in kinesis and I don't know if it is in the impl for s3 auth.

**Comment by xxchan** [comment]
ping @BugenZhao @tabVersion, can we merge?

**Review by BugenZhao — APPROVED** [review]
I'm fine with the refactor itself. We may seek advice from @tabVersion for the actual business.

File: src/connector/with_options_source.yaml [hunk 4]
```diff
@@ -455,9 +455,31 @@ PulsarProperties:
...
+    required: false
+  - name: session_token
+    field_type: String
+    required: false
+  - name: arn
+    field_type: String
+    required: false
+  - name: external_id
+    field_type: String
+    comments: This field was added for kinesis. Not sure if it's useful for other connectors.  Please ignore it in the documentation for now.
```
Comment (tabVersion) [inline src/connector/with_options_source.yaml:479]: `external_id` serves as a common part for aws auth, not only related to kinesis but also s3. 
I am also drafting a doc for aws auth usage, demonstrating how these fields work together.

**Review by tabVersion — APPROVED** [review]
LGTM

**Pull request merged**